apis/python/src/tiledbvcf/binding/reader.cc

```diff
@@ -401,6 +408,28 @@ std::vector<std::string> Reader::get_queryable_attributes() {
   return attrs;
 }
 
+int32_t Reader::get_sample_count() {
+  auto reader = ptr.get();
+  int32_t count;
+  check_error(reader, tiledb_vcf_reader_get_sample_count(reader, &count));
+  return count;
+}
+
+std::vector<std::string> Reader::get_sample_names() {
+  auto reader = ptr.get();
+  int32_t count;
+  std::vector<std::string> names;
+  check_error(reader, tiledb_vcf_reader_get_sample_count(reader, &count));
+
```

Comment:
Add a `names.reserve(count);` here so that it pre-allocates exactly enough space for the names that we'll add. Otherwise this may realloc many times, possibly over-allocating memory.